File: apps/home-service-platform-server/src/APIs/Booking/Dtos/BookingWhereInput.cs
namespace HomeServicePlatform.APIs.Dtos;

public class BookingWhereInput
{
    public DateTime? BookingDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? Id { get; set; }

    public string? PaymentStatus { get; set; }

    public string? ServiceCategory { get; set; }

    public string? ServiceProvider { get; set; }

    public List<string>? ServiceProviders { get; set; }

    public string? Status { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? User { get; set; }

    public List<string>? Users { get; set; }
}

File: apps/home-service-platform-node-js-admin/src/api/booking/BookingWhereInput.ts
import { AdminListRelationFilter } from "../admin/AdminListRelationFilter";
import { DateTimeNullableFilter } from "../../util/DateTimeNullableFilter";
import { StringFilter } from "../../util/StringFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { ServiceCategoryWhereUniqueInput } from "../serviceCategory/ServiceCategoryWhereUniqueInput";
import { ServiceProviderWhereUniqueInput } from "../serviceProvider/ServiceProviderWhereUniqueInput";
import { ServiceProviderListRelationFilter } from "../serviceProvider/ServiceProviderListRelationFilter";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";

export type BookingWhereInput = {
  admins?: AdminListRelationFilter;
  bookingDate?: DateTimeNullableFilter;
  id?: StringFilter;
  paymentStatus?: StringNullableFilter;
  serviceCategory?: ServiceCategoryWhereUniqueInput;
  serviceProvider?: ServiceProviderWhereUniqueInput;
  serviceProviders?: ServiceProviderListRelationFilter;
  status?: StringNullableFilter;
  user?: UserWhereUniqueInput;
};
